plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -177,7 +178,36 @@ private Stream<DeltaLakeSplit> getSplits(
 
         MetadataEntry metadataEntry = tableHandle.getMetadataEntry();
         boolean isOptimize = tableHandle.isOptimize();
-        Set<Map<String, Optional<String>>> partitionsWithAtMostOneFile = isOptimize ? findPartitionsWithAtMostOneFile(validDataFiles) : ImmutableSet.of();
+        if (isOptimize) {
+            Map<Map<String, Optional<String>>, List<AddFileEntry>> queuedAddFileEntriesMap = new HashMap<>();
```

Comment:
@homar  FYI this is a slight rework of our initial fix from https://github.com/trinodb/trino/pull/18938